### DISCUSSION TITLE
How to forbid deletion

### DISCUSSION THREAD

**Discussion by super-kamil** [strapi/strapi#7011] [Legacy Topics]
Hi, 

is there any way to forbid deletion of an item in strapi admin?

![image](https://user-images.githubusercontent.com/708028/87431618-2fe6e600-c5e7-11ea-8117-de6b18bb4011.png)

Example: website categories can be edited in strapi backend but it's not allowed to delete them because they are important for the whole system.

In 3.x-beta the beforeDestroy lifecycle callback worked by throwing an error there but now this does not prevent the deletion anymore. Any ideas? 

Thanks!

**Comment by goxiaoy** [comment]
create a model `category` with field `preserved`
```
  "attributes": {
    ...
    "preserved": {
      "type": "boolean",
      "configurable": false,
      "private": true
    }
```

update `api/category/models/category.js`
```
'use strict';

/**
 * Lifecycle callbacks for the `category` model.
 */

module.exports = {
  lifecycles: {
    async beforeUpdate(params,data) {
      if(data.preserved){
        throw strapi.errors.badRequest('Perserved');
      }
    },
    async beforeDelete(params){
      params.preserved = false;
      console.log(params);
    }
  }
};

```

this will prevent category being deleted or updated

**Comment by lauriejim** [comment]
Hello! This is possible to have that in the Strapi Enterprise Edition.
Administrators have roles in the 3.1.x version of Strapi and in the CE (Community Edition) you can just limit interaction on the Content Type level and in the EE (Enterprise Edition) you can customize action (create read update delete) and fields permissions.

After that, you can try to customize the admin panel using the Customization concept if you want to remove this button for a specific Content Type.

**Comment by super-kamil** [comment]
Thank you! I will test this.
